lib/PuppeteerSharp.Tests/Network/NetworkEventTests.cs

```diff
@@ -0,0 +1,177 @@
+﻿using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace PuppeteerSharp.Tests.Network
+{
+    [Collection("PuppeteerLoaderFixture collection")]
+    public class NetworkEventTests : PuppeteerPageBaseTest
+    {
+        [Fact]
+        public async Task PageEventsRequest()
+        {
+            var requests = new List<Request>();
+            Page.RequestCreated += (sender, e) => requests.Add(e.Request);
+            await Page.GoToAsync(TestConstants.EmptyPage);
+            Assert.Single(requests);
+            Assert.Equal(TestConstants.EmptyPage, requests[0].Url);
+            Assert.Equal(ResourceType.Document, requests[0].ResourceType);
+            Assert.Equal("GET", requests[0].Method);
+            Assert.NotNull(requests[0].Response);
+            Assert.Equal(Page.MainFrame, requests[0].Frame);
+            Assert.Equal(TestConstants.EmptyPage, requests[0].Frame.Url);
+        }
+
+        [Fact]
+        public async Task PageEventsRequestShouldReportPostData()
+        {
+            await Page.GoToAsync(TestConstants.EmptyPage);
+            Server.SetRoute("/post", context => Task.CompletedTask);
+            Request request = null;
+            Page.RequestCreated += (sender, e) => request = e.Request;
+            await Page.EvaluateExpressionHandleAsync("fetch('./post', { method: 'POST', body: JSON.stringify({ foo: 'bar'})})");
+            Assert.NotNull(request);
+            Assert.Equal("{\"foo\":\"bar\"}", request.PostData);
+        }
+
+        [Fact]
+        public async Task PageEventsResponse()
+        {
+            var responses = new List<Response>();
+            Page.ResponseCreated += (sender, e) => responses.Add(e.Response);
+            await Page.GoToAsync(TestConstants.EmptyPage);
+            Assert.Single(responses);
+            Assert.Equal(TestConstants.EmptyPage, responses[0].Url);
+            Assert.Equal(HttpStatusCode.OK, responses[0].Status);
+            Assert.NotNull(responses[0].Request);
+        }
+
+        [Fact]
+        public async Task PageEventsResponseShouldProvideBody()
+        {
+            Response response = null;
+            Page.ResponseCreated += (sender, e) => response = e.Response;
+            await Page.GoToAsync(TestConstants.ServerUrl + "/simple.json");
+            Assert.NotNull(response);
+            Assert.Equal($"{{\"foo\": \"bar\"}}{Environment.NewLine}", await response.TextAsync());
```

Comment:
fails:
```ini
Assert.Equal() Failure
                        ↓ (pos 14)
Expected: {"foo": "bar"}\r\n
Actual:   {"foo": "bar"}\n
                        ↑ (pos 14)
```
not sure how to work around this, works locally (windows 10)

Reply:
I had the same issue in other test. I ended up using `\n` instead of NewLine. I'll work on Mac

Reply:
on windows it would fail:
```cs
Assert.Equal("{\"foo\": \"bar\"}\n", await response.TextAsync());
```

```ini
Result Message:	
Assert.Equal() Failure
                        ↓ (pos 14)
Expected: {"foo": "bar"}\n
Actual:   {"foo": "bar"}\r\n
                        ↑ (pos 14)


```

Reply:
I'm trying to give you a good solution without trying to sound lazy :p  

Could you try instead of hardkoding the json text. open the simple.json file?
So the test would be about "You are sending me the same content as the file I'm requesting".